R/Data-methods.R

```diff
@@ -25,91 +25,93 @@ setMethod(
-                cohort.id <- unique(x@cohort)
-                for(iCoh in seq(a=cohort.id)){
-                  filter.coh <- which(x@cohort == cohort.id[iCoh])
-                  df[filter.coh,"dose"] <- max(df[filter.coh,"dose"])
-                  df[filter.coh,"toxicity"] <- sort(df[filter.coh,"toxicity"],
-                                                    decreasing=TRUE)
-                }
-              }
-
-              a <- ggplot(df, aes(x=patient,y=dose)) +
-                  scale_y_continuous(breaks=
-                                     sort(unique(c(0, df$dose))),
-                                     minor_breaks=numeric(),
-                                     limits=c(0, max(df$dose) * 1.1))
-
-              a <- a +
-                  geom_point(aes(shape=toxicity,colour=toxicity),
-                             size=3) +
-                                 scale_colour_manual(values=cols) +
-                                     xlab("Patient") + ylab("Dose Level")
-
-              if(!blind)
-                a <- a + geom_text(aes(label=ID, size=2),
-                                   data=df,
-                                   hjust=0, vjust=0.5,
-                                   angle=90, colour=I("black"),
-                                   show.legend = FALSE)
-
-              a <- a + scale_x_continuous(breaks=df$patient,
-                                          minor_breaks=numeric())
-
-              # add a vertical lines separating sub-sequent cohorts
-              if(x@placebo & length(unique(x@cohort)) > 1)
-                a <- a + geom_vline(xintercept=head(cumsum(table(x@cohort)),n=-1) + 0.5,
-                                    colour="green",
-                                    linetype = "longdash")
-
-              return(a)
-          })
+# Data-plot ----
 
+#' Plot method for the [`Data`] class.
+#'
+#' @description `r lifecycle::badge("stable")`
+#'
+#' A method that creates a plot for [`Data`] object.
+#'
+#' @param x (`GeneralData`)\cr object we want to plot.
+#' @param blind (`flag`)\cr indicates whether to blind the data.
+#'   If `TRUE`, then placebo subjects are reported at the same level
+#'   as the active dose level in the corresponding cohort,
+#'   and DLTs are always assigned to the first subjects in a cohort.
+#' @param \dots not used.
+#'
+#' @return The [`ggplot2`] object.
+#'
+#' @name plot-Data-method
+#' @example examples/Data-method-plot-Data.R
+#' @export
+#'
+setMethod(
+  f = "plot",
+  signature = signature(x = "Data", y = "missing"),
+  def = function(x, y, blind = FALSE, ...) {
+    if (x@nObs == 0L) {
+      return()
+    }
+
+    if (blind) {
+      x <- h_data_blind(x)
+    } else if (x@placebo) {
+      # Placebo will be plotted at y = 0 level.
+      x@x[x@x == x@doseGrid[1]] <- 0
+    }
+
+    df <- data.frame(
+      patient = seq_along(x@x),
+      ID = paste(" ", x@ID),
+      dose = x@x,
+      toxicity = as.factor(x@y),
+    )
```

Comment:
please create another helper `h_plot_data_df()` for this, such that you can just call here `df <- h_plot_data_df(x, blind)`. 

For the blinding, I would not do that in `x` directly, the reason is that we don't have a way to signal that this object has been blinded. So I would prefer to keep that at a lower level instead and just do this in the `df`.